File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import sys
input = sys.stdin.readline

def eratos(n, sosu):
    for i in range(2, 2*n+1):
        for j in range(i*2, 2*n+1, i):
            if sosu[j]:
                sosu[j] = False


def solution(n):
    sosu = [True] * (2*n+1)
    sosu[0], sosu[1] = False, False

    eratos(n, sosu)
    ans = 0

    for i in range(n+1, 2*n+1):
        if sosu[i]:
            ans += 1
    
    return ans

while True:
    n = int(input())
    if n == 0:
        break

    ans = solution(n)
    print(ans)

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.cc
#include <iostream>

using namespace std;

const int MAX = 123456 * 2;

int main()
{
	int n;
	while (true) {
		cin >> n;
		if (n == 0) break;
		
		int sosu[MAX] = {0};
		sosu[0] = 1;
		sosu[1] = 1;
		
		int ans = 0;
		// 에라토스테네스의 체
		for (int i = 2; i <= 2*n; i++) {
			for (int j = i*2; j <= 2*n; j += i) {
				if (sosu[j] == 0) sosu[j] = 1;
			}
		}

		for (int i = n+1; i <= 2*n; i++) if (sosu[i] == 0) ans++;

		cout << ans << endl;
	}

	return 0;
}